demos/storybook/stories/user-menu/with-basic-usage.tsx

```diff
@@ -0,0 +1,42 @@
+import { Avatar } from '@material-ui/core';
+import { Email, ExitToApp, Settings } from '@material-ui/icons';
+import { UserMenu, UserMenuItem } from '@pxblue/react-components';
+import { action } from '@storybook/addon-actions';
+import { text } from '@storybook/addon-knobs';
+import { StoryFnReactReturnType } from '@storybook/react/dist/client/preview/types';
+import React from 'react';
+
+const menuItems: UserMenuItem[] = [
+    {
+        itemID: '1',
+        title: 'Account',
+        icon: <Settings />,
+        onClick: action("click 'Account'"),
+    },
+    {
+        itemID: '2',
+        title: 'Contact Us',
+        icon: <Email />,
+        onClick: action("click 'Contact Us'"),
+    },
+    {
+        itemID: '3',
+        title: 'Log Out',
+        icon: <ExitToApp />,
+        onClick: action("click 'Log Out'"),
+    },
+];
+
+export const menuGroups = [
+    {
+        items: menuItems,
+    },
+];
+
+export const withBasicUsage = (): StoryFnReactReturnType => {
+    const value = text('Avatar.value', 'AB');
+    const avatar = <Avatar>{value}</Avatar>;
+    return <UserMenu avatar={avatar} menuGroups={menuGroups} onOpen={action('open')} onClose={action('close')} />;
+};
```

Comment:
No not everything is required; the onOpen and onClose are def not required.  I think it would be good to include those events though in every story so that the actions tab works consistently.  

MenuGroups is marked as an optional param, but it doesn't make a lot of sense to me to have them as optional.  I suggest we make menuGroups a required prop, because otherwise the user should just be using an Avatar, not a UserMenu.

Reply:
Yeah, let's make it required